xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/doc/XWikiDocument.java

```diff
@@ -9237,4 +9237,17 @@ private XWikiAttachmentStoreInterface getXWikiAttachmentStoreInterface(String st
 
         return null;
     }
+
+    /**
+     * Compute and return the maximum authorized length for the full name (i.e. the serialized reference of the
+     * document) based on the current store limitation.
+     *
+     * @return the maximum authorized length for a document full name.
+     * @since 11.3RC1
+     */
+    @Unstable
+    public int getFullNameMaxLength()
```

Comment:
As I said in my previous comment I think `getLocalReferenceMaxLength` would be better since getFullName() is deprecated.

Reply:
Hmm indeed sorry, I saw the comment before you edited it :)